examples/typescript-simplechat/client.ts

```diff
@@ -0,0 +1,78 @@
+import * as readline from "readline";
+
+const model = "llama2";
+type Message = {
+  role: "assistant" | "user" | "system";
+  content: string;
+}
+const messages: Message[] = [{
+  role: "system",
+  content: "You are a helpful AI agent."
+}]
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+})
+
+async function chat(messages: Message[]): Promise<Message> {
+  const body = {
+    model: model,
+    messages: messages
+  }
+
+  const response = await fetch("http://localhost:11434/api/chat", {
+    method: "POST",
+    body: JSON.stringify(body)
+  })
+
+  const reader = response.body?.getReader()
+  if (!reader) {
+    throw new Error("Failed to read response body")
+  }
+  const content: string[] = []
+  while (true) {
+    const { done, value } = await reader.read()
+    if (done) {
+      break;
+    }
+    const rawjson = new TextDecoder().decode(value);
+    const json = JSON.parse(rawjson)
+
+    if (json.done === false) {
+      process.stdout.write(json.message.content);
+      content.push(json.message.content)
+      // messages.push({role: "system", content: text})
+    }
+
+  }
+  return { role: "assistant", content: content.join("") };
+}
+
+async function askQuestion(): Promise<void> {
+  return new Promise<void>((resolve) => {
+    rl.question("\n\nAsk a question: (press enter alone to quit)\n\n", async (user_input) => {
+      if (user_input.trim() === "") {
+        rl.close();
+        console.log("Thankyou. Goodbye.\n")
+        console.log("=======\nHere is the message history that was used in this conversation.\n=======\n")
+        messages.forEach(message => {
```

Comment:
You can just log the messages array too and it will output nicely. This is also fine though.

Here's what just logging messages looks like:
```
[
  { role: 'system', content: 'You are a helpful AI agent.' },
  { role: 'user', content: 'hi' },
  {
    role: 'assistant',
    content: "Hello! It's nice to meet you. I'm here to help with any questions or tasks you may have, how can I assist you today? Please let me know if there is anything specific you need help with."
  },
  { role: 'user', content: 'what is up?' }
]
```